### DISCUSSION TITLE
How to protect jsBundle

### DISCUSSION THREAD

**Discussion by elmcapp** [react-native-community/discussions-and-proposals#547] [Ideas]
So I created a catalyst app with React Native. One issue is that my main.jsBundle is expose. users can easy go to the app/content/resource an have access to bundle. Is there a way I can protect this file from being expose or stealing my source code

**Comment by radko93** [comment]
You can see the bundle of any React Native Android app by opening the apk file. So you cannot really hide the file.

**Reply by elmcapp** [reply]
So I have a macOS Catalyst app using React Native. The fact that the bundle is exposed
1. does this mean that my app would be able to be hacked. 
2. Would a user be able to change some of the values to get access to paid restricted features in the app
3. Would someone be able to copy my app by using my main js bundle?
Just trying to make sure a lot of my hard work is not able to be stolen simply by the main js bundle being available 


![Screen Shot 2022-11-17 at 2 33 47 PM](https://user-images.githubusercontent.com/64275658/202541774-1456b64a-aa0d-4c35-a2ac-391c847445f6.png)

**Reply by radko93** [reply]
You can open the bundle using a code editor and see if the code is readable. If it's not readable then using it might be very difficult for someone else.

**Comment by cortinico** [comment]
You will have to use a "Javascript Obfuscator" tool like DexGuard:
https://www.guardsquare.com/blog/obfuscating-javascript-code-dexguard-and-ixguard
to achieve what you're looking for.

**Comment by Kudo** [comment]
the alternative way is probably to use hermes. hermes will convert the plain text js bundle to bytecode bundle. that would increase the difficulty to trace your source code.
note that there is no perfect security. bytecode or even binary could both be disassembled.
